### PR TITLE
Add explicit support for py3.5

### PR DESCRIPTION
**Pull request by ticosax**
Hi,
Since python3.5 works out of the box, I thought it would be useful to advertise it.

### CODE REVIEW

File: cli/tox.ini [hunk 1]
```diff
@@ -10,7 +10,7 @@ deps =
   -e..
 passenv = DCOS_* CI_FLAGS CLI_TEST_SSH_KEY_PATH CLI_TEST_MASTER_PROXY
 
-[testenv:py34-syntax]
```
Comment (tamarrow) [inline cli/tox.ini]: We should keep syntax checks in one python version, or else we'll have conflicts :)

**Comment by tamarrow** [comment]
@ticosax Thanks for contributing! I'm not sure that all features work in python 35, but if all the tests pass will merge! Looks like we need to require a higher version of tox to include py35 support in `requirements.txt`

File: cli/tox.ini (hunk 1 above)
Comment (ticosax) [inline cli/tox.ini]: If you prefer, I think it is better to check with latest supported python version to catch `async` keyword used as a variable, for instance,

```python
def foo(async=True):
    pass
```
This snippet will pass with python3.4 but not with python3.5

**Comment by ticosax** [comment]
> Looks like we need to require a higher version of tox to include py35 support in requirements.txt

According to what I [read](https://teamcity.mesosphere.io/viewLog.html?buildId=306305&buildTypeId=DcosIo_DcosCli_Ci&tab=buildLog). This is not a `tox` issue, it is because python3.5 is not available in testing environment.

**Comment by tamarrow** [comment]
ok to test

**Comment by tamarrow** [comment]
@ticosax I'm going to close this pr because it requires updating the CIs and we already have an issue to upgrade our supported version of python to 35: #766 

If you disagree, feel free to re-open the pr.

**Pull request closed without merging**